src/main/java/com/example/wdmsystem/util/JwtAuthTokenFilter.java

```diff
@@ -38,13 +41,15 @@ protected void doFilterInternal(HttpServletRequest request, HttpServletResponse
                 Claims claims = jwtUtil.validateToken(jwt);
                 String username = claims.getSubject();
                 List<String> roles = parseRoles(claims.get("roles"));
-                Integer merchantId = claims.get("merchantId", Integer.class);
-
+                //Integer merchantId = claims.get("merchantId", Integer.class);
+                Merchant merchant = claims.get("merchantId", Merchant.class); // 
```

Comment:
Here and in JWtUtil it should be kept as was, storing the whole entity in the token is not really safe